gridfs/__init__.py

```diff
@@ -21,980 +21,46 @@
 """
 from __future__ import annotations
 
-from collections import abc
-from typing import Any, Mapping, Optional, cast
-
-from bson.objectid import ObjectId
+import sys
+
+from gridfs.asynchronous.grid_file import (
+    AsyncGridFS,
+    AsyncGridFSBucket,
+    AsyncGridIn,
+    AsyncGridOut,
+    AsyncGridOutCursor,
+)
 from gridfs.errors import NoFile
-from gridfs.grid_file import (
-    DEFAULT_CHUNK_SIZE,
+from gridfs.grid_file_shared import DEFAULT_CHUNK_SIZE
+from gridfs.synchronous.grid_file import (
+    GridFS,
+    GridFSBucket,
     GridIn,
     GridOut,
     GridOutCursor,
-    _clear_entity_type_registry,
-    _disallow_transactions,
 )
-from pymongo import ASCENDING, DESCENDING, _csot
-from pymongo.client_session import ClientSession
-from pymongo.collection import Collection
-from pymongo.common import validate_string
-from pymongo.database import Database
-from pymongo.errors import ConfigurationError
-from pymongo.read_preferences import _ServerMode
-from pymongo.write_concern import WriteConcern
+
+# Export synchronous modules as top-level gridfs modules for compatibility
+_sync_modules = {}
+for _name in sys.modules:
+    if _name.startswith("gridfs.synchronous."):
+        _full_name = "{}.{}".format("gridfs", _name.rsplit(".")[-1])
+        _sync_modules[_full_name] = _name
+
+for _module in _sync_modules:
+    sys.modules[_module] = sys.modules[_sync_modules[_module]]
```

Comment:
Sounds good but I want to clarify that this issue doesn't need to block this PR. It would be fine to open a ticket to change this later if that's easier.

Reply:
It's easy enough to do now and get the structure of the repo into a correct state.